commands/addons/open.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+let cli = require('heroku-cli-util');
+let co  = require('co');
+let opn = require('opn');
+let fs  = require('fs');
+
+function open (url) {
+  cli.log(`Opening ${cli.color.cyan(url)}...`);
+  return new Promise(function (fulfill, reject) {
+    opn(url, {wait: false}, function (err) {
+      if (!err) return fulfill();
+      cli.error('Cannot open page.');
+      cli.error(err);
+      reject(new Error(`Open ${cli.color.cyan(url)} in your browser.`));
+    });
+  });
+}
+
+const ssoPath = '/tmp/heroku-sso.html';
+
+function writeTemplate (ctx, sso, path) {
```

Comment:
updated.

Yeah I wasn't super happy with how this works either, but it does, and seems secure. It's how it worked before pretty much exactly.